lib/ROM/Rom.cpp

```diff
@@ -0,0 +1,135 @@
+// ****************************************************************************** //
+//   _______   __                              __       __                        //
+//  /       \ /  |                            /  \     /  |                       //
+//  $$$$$$$  |$$ |  ______   __    __         $$  \   /$$ |  ______   _______     //
+//  $$ |__$$ |$$ | /      \ /  |  /  | ______ $$$  \ /$$$ | /      \ /       \    //
+//  $$    $$/ $$ | $$$$$$  |$$ |  $$ |/      |$$$$  /$$$$ | $$$$$$  |$$$$$$$  |   //
+//  $$$$$$$/  $$ | /    $$ |$$ |  $$ |$$$$$$/ $$ $$ $$/$$ | /    $$ |$$ |  $$ |   //
+//  $$ |      $$ |/$$$$$$$ |$$ \__$$ |        $$ |$$$/ $$ |/$$$$$$$ |$$ |  $$ |   //
+//  $$ |      $$ |$$    $$ |$$    $$ |        $$ | $/  $$ |$$    $$ |$$ |  $$ |   //
+//  $$/       $$/  $$$$$$$/  $$$$$$$ |        $$/      $$/  $$$$$$$/ $$/   $$/    //
+//                          /  \__$$ |                                            //
+//                          $$    $$/                                             //
+//                           $$$$$$/                                              //
+//                                                                                //
+//                            By: K1ngmar and rvan-mee                            //
+// ****************************************************************************** //
+
+#include <play-man/ROM/Rom.hpp>
+#include <iostream>
+#include <iomanip>
+#include <string_view>
+#include <type_traits>
+#include <optional>
+#include <ios>
+
+namespace GameBoy {
+
+    Rom::Rom(const char* filePath)
+    {
+        _filePath = filePath;
+        ParseRomFile(filePath);
+    }
+}
+
+
+static int32_t GetPaddingAfterName(const std::string_view name)
+{
+    constexpr size_t headerPrintLineWidth = 52;
+    constexpr size_t decorationWidth = (sizeof("/     ")-1) + (sizeof(": ")-1) + (sizeof("     \\")-1);
+    const size_t padding = headerPrintLineWidth - (decorationWidth + name.size());
+
+    return (padding);
+}
+
+template<class T>
+static void PrintIntegralLine(const std::string_view name, T var, int32_t bytesInHexValue)
+{
+    int32_t padding = GetPaddingAfterName(name) - (bytesInHexValue * 2);
+    auto resetFlags = std::cout.flags();
+    std::cout << std::hex << std::uppercase;
+
+    // print the starting decoration and the name
+    std::cout << "/     " << name << ": ";
+    // offset the start of the hex by as many bytes needed for proper formatting
+    std::cout << std::setw(padding) << "0x";
+    // print the value as a hex, padded by 0's
+    std::cout << std::setfill('0') << std::setw(bytesInHexValue * 2) << var;
+    // print the trailing decoration
+    std::cout << "     \\" << std::endl;
+
+    // reset the cout formatting
+    std::cout.flags(resetFlags);
+    std::cout << std::setfill(' ');
+}
+
+static int32_t ConvertInt8(int8_t val)
+{
+    return (static_cast<int32_t>(val) & 0xFF);
```

Comment:
IIRC the & 0xFF was necessary for negative values. I think it is cleaner to use a uint8_t for raw byte values as they have no concept of negative meaning we wouldnt have to do stuff like `& 0xFF`